app/src/main/java/com/android/sample/ui/authentication/SignUp.kt

```diff
@@ -0,0 +1,315 @@
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawWithCache
+import androidx.compose.ui.graphics.BlendMode
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.R
+import com.android.sample.ui.theme.PrimaryGradientBrush
+import com.android.sample.ui.theme.PrimaryPurple
+import com.android.sample.ui.theme.PrimaryRed
+import com.android.sample.ui.theme.SecondaryPurple
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SignUpScreen() {
+  Scaffold(
+      modifier = Modifier.testTag("signUpScreen"),
+      topBar = {
+        TopAppBar(
+            title = {
+              Box(
+                  modifier = Modifier.fillMaxWidth().padding(end = 36.dp),
+                  contentAlignment = Alignment.Center) {
+                    Text(
+                        modifier = Modifier.testTag("appName"),
+                        text =
+                            buildAnnotatedString {
+                              append("Beat")
+                              withStyle(style = SpanStyle(color = PrimaryRed)) { append("Link") }
+                            },
+                        style =
+                            TextStyle(
+                                fontSize = 20.sp,
+                                fontFamily = FontFamily(Font(R.font.roboto)),
+                                fontWeight = FontWeight(700),
+                                color = PrimaryPurple,
+                                letterSpacing = 0.2.sp,
+                                textAlign = TextAlign.Center))
```

Comment:
We could create these text styles into Theme so we don't find these repeated line of codes, but this is ok for this PR, will be implemented next week.